ci/helm-chart/templates/deployment.yaml

```diff
@@ -21,6 +21,10 @@ spec:
         app.kubernetes.io/name: {{ include "code-server.name" . }}
         app.kubernetes.io/instance: {{ .Release.Name }}
     spec:
+      {{- with .Values.imagePullSecrets }}
```

Comment:
Thanks so much for your contribution! I have a small suggestion, but it's a style nitpick, so feel free to ignore it, since we can always address it later.

The following code should produce equivalent behavior, and is easier to read (in my completely subjective opinion) than the current approach:

```yaml
    spec:
      imagePullSecrets: {{ toYaml .Values.imagePullSecrets | nindent 8 }}
```

I tested this with the default empty array value, and it renders the following output:

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: code-server
spec:
  template:
    spec:
      imagePullSecrets: 
        []
```

You can try this yourself using `helm template code-server .` from the `helm-chart` directory.

With the example, this produces the following output:

```yaml
    spec:
      imagePullSecrets: 
        - name: registry-creds
```

I have not tested this with an actual deployment. We might want to consider adding an automated test that deploys code-server into a `kind` cluster to verify that an actual deploy continues working as we make modifications to this chart, but we can do that separately.

Reply:
Yep, understood! My comment was not a criticism of what we had done before, just something for us to consider in the future, so that we can be confident about changes that we're making to the Helm chart. I'm working on getting the chart published automatically, so automating checks/tests will be an important prerequisite so that we don't accidentally push bad charts

Reply:
Absolutely! I know it was not a criticism, I just wanted to make sure you had to the backstory from the initial merge :3